datastore/__init__.py

```diff
@@ -661,6 +666,87 @@ async def get_seed_info(cursor):
         seed_dict["seeds"].append({"seed_id": seed_id, "seed_name": seed_name})
     return seed_dict
 
+async def get_picture_sets_info(cursor, user_id: str):
+    """This function retrieves the picture sets names and number of pictures from the database.
+
+    Args:
+        user_id (str): id of the user
+    """
+    # Check if user exists
+    if not user.is_a_user_id(cursor=cursor, user_id=user_id):
+        raise user.UserNotFoundError(
+            f"User not found based on the given id: {user_id}"
+        )
+    
+    result = {}
+    picture_sets = picture.get_user_picture_sets(cursor, user_id)
+    for picture_set in picture_sets:
+        picture_set_id = picture_set[0]
+        picture_set_name = picture_set[1]
+        nb_picture = picture.count_pictures(cursor, picture_set_id)
+        result[picture_set_id] = [picture_set_name, nb_picture]
+    return result
+
+async def delete_picture_set(cursor, user_id, picture_set_id, container_client):
+    """
+    Delete a picture set from the database and the blob storage
+
+    Args:
+        cursor: The cursor object to interact with the database.
+        user_id (str): id of the user
+        picture_set_id (str): id of the picture set to delete
+        container_client: The container client of the user.
+    """
+    try:
+        # Check if user exists
+        if not user.is_a_user_id(cursor=cursor, user_id=user_id):
+            raise user.UserNotFoundError(
+                f"User not found based on the given id: {user_id}"
+            )
+        # Check if picture set exists
+        if not picture.is_a_picture_set_id(cursor, picture_set_id):
+            raise picture.PictureSetNotFoundError(
+                f"Picture set not found based on the given id: {picture_set_id}"
+            )
+        # Check user is owner of the picture set
+        if not picture.get_picture_set_owner_id(cursor, picture_set_id) == user_id:
+            raise picture.PictureSetDeleteError(
+                f"User can't delete this folder, user uuid :{user_id}, folder name : {picture_set_id}"
+            )
+        # Check if the picture set is the default picture set
+        general_folder_id = user.get_default_picture_set(cursor, user_id)
+        if general_folder_id == picture_set_id:
+            raise picture.PictureSetDeleteError(
+                f"User can't delete the default picture set, user uuid :{user_id}"
+            )
+
+        if picture.count_pictures(cursor, picture_set_id) > 0:
+            #move the pictures to the default picture set	
+            for p in picture.get_picture_set_pictures(cursor, picture_set_id):
+                picture_id = p[0]
+                # set picture set to default one
+                picture.update_picture_picture_set_id(cursor, picture_id, general_folder_id)
```

Comment:
wouldn't it be more efficient to:
```SQL
UPDATE TABLE X SET picture_set_id = value_of_new_picture_set_id 
WHERE picture_set_id=old_value_of_new_picture_set_id
```
Therefore we wont miss any entries in the DB?